Extract/Extract.ipynb

```diff
@@ -0,0 +1,264 @@
+    "            uuid = UUID(uuidstr)\n",
+    "            ctypes.Structure.__init__(self)\n",
+    "            self.Data1, self.Data2, self.Data3, \\\n",
+    "                self.Data4[0], self.Data4[1], rest = uuid.fields\n",
+    "            for i in range(2, 8):\n",
+    "                self.Data4[i] = rest>>(8-i-1)*8 & 0xff\n",
+    "\n",
+    "    SHGetKnownFolderPath = windll.shell32.SHGetKnownFolderPath\n",
+    "    SHGetKnownFolderPath.argtypes = [\n",
+    "        ctypes.POINTER(GUID), wintypes.DWORD,\n",
+    "        wintypes.HANDLE, ctypes.POINTER(ctypes.c_wchar_p)\n",
+    "    ]\n",
+    "\n",
+    "    def _get_known_folder_path(uuidstr):\n",
+    "        pathptr = ctypes.c_wchar_p()\n",
+    "        guid = GUID(uuidstr)\n",
+    "        if SHGetKnownFolderPath(ctypes.byref(guid), 0, 0, ctypes.byref(pathptr)):\n",
+    "            raise ctypes.WinError()\n",
+    "        return pathptr.value\n",
+    "\n",
+    "    FOLDERID_Download = '{374DE290-123F-4565-9164-39C4925E467B}'\n",
+    "\n",
+    "    def get_download_folder():\n",
+    "        return _get_known_folder_path(FOLDERID_Download)\n",
+    "else:\n",
+    "    def get_download_folder():\n",
+    "        home = os.path.expanduser(\"~\")\n",
+    "        return os.path.join(home, \"Downloads\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "###Get position of annotations in text (Mat take a while for large files). Will be in downloads folder"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The total number of matches is 24\n"
+     ]
+    }
+   ],
+   "source": [
+    "'''\n",
+    "If total is off uncomment lines to check why\n",
+    "'''\n",
+    "root, ext = os.path.splitext(Path(file_path).name)\n",
+    "name = get_download_folder()\n",
+    "name = name + \"/\" + root + '.csv'\n",
+    "#TODO change 1522.csv to use identifier\n",
+    "#p = urllib.parse.urljoin(name, '1522.csv')\n",
+    "count = 0\n",
+    "with open(name, \"w\") as csv_file:\n",
+    "    writer = csv.writer(csv_file, delimiter=',')\n",
+    "    header = [\"Segment\", \"Start\", \"End\", \"Occurs\", \"Concept\"]\n",
+    "    writer.writerow(header)\n",
+    "    for index, row in df.iterrows():\n",
+    "        l = []\n",
+    "        seg = row['Segment']\n",
+    "        seg = re.sub(r'\\(', r'\\\\(', seg ) #Need to escape paraenthesis due to regex\n",
+    "        seg = re.sub(r'\\)', r'\\\\)', seg )\n",
+    "        seg = re.sub(r'\\?', r' ', seg )\n",
+    "        seg = re.sub(r'[^\\x00-\\x7F]+',' ', seg)\n",
+    "        seg = re.sub(r'\\s+', ' ', seg)\n",
+    "        seg = re.sub(r'\\\\r', ' ', seg)\n",
+    "        m = re.search(seg, text, re.M | re.DOTALL)\n",
+    "        url =  'http://diging.asu.edu/uri/leuphana' + str(row['Dokumentname'])\n",
```

Comment:
Isn't there a slash at the end of the prefix missing?

Reply:
must have left that out when I changed it to leuphana, fixed